Mathlib/Algebra/Category/Ring/Topology.lean

```diff
@@ -0,0 +1,176 @@
+/-
+Copyright (c) 2025 Andrew Yang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Andrew Yang, Christian Merten
+-/
+import Mathlib.Algebra.Category.Ring.Colimits
+import Mathlib.Algebra.Category.Ring.Constructions
+import Mathlib.Algebra.MvPolynomial.CommRing
+import Mathlib.Topology.Algebra.Ring.Basic
+
+/-!
+# Topology on `Hom(R, S)`
+
+In this file, we define topology on `Hom(R, S)` for a topological ring `S`,
+given by the coarsest topology that makes `f ↦ f x` continuous for all `x : R`.
+Alternatively, given a presentation `R = ℤ[x₁,...,xₙ]/I`,
+this is the subspace topology `Hom(R, S) ↪ Hom(ℤ[x₁,...,xₙ], S) = Sⁿ`.
+
+## Main results
+- `CommRingCat.HomTopology.isClosedEmbedding_precomp_of_surjective`:
+  `Hom(R/I, T)` is a closed subspace of `Hom(R, T)` if `T` is Hausdorff.
+- `CommRingCat.HomTopology.mvPolynomialHomeo`:
+  `Hom(S[Xᵢ], R)` is homeomorphic to `Hom(S, R) × Rⁱ`.
+- `CommRingCat.HomTopology.isEmbedding_pushout`:
+  `Hom(A ⊗[S] B, R)` has the subspace topology from `Hom(A, R) × Hom(B, R)`.
+
+-/
+
+universe u v
+
+open CategoryTheory Topology
+
+namespace CommRingCat.HomTopology
+
+variable {A B R S T : CommRingCat.{u}}
+
+/--
+The topology on `Hom(R, S)` for a topological ring `S`, given by the coarsest topology that
+makes `f ↦ f x` continuous for all `x : R` (see `continuous_apply`).
+Alternatively, given a presentation `R = ℤ[x₁,...,xₙ]/I`,
+This is the subspace topology `Hom(R, S) ↪ Hom(ℤ[x₁,...,xₙ], S) = Sⁿ` (see `mvPolynomialHomeo`).
+
+This is a scoped instance in `CommRingCat.HomTopology`.
+-/
+scoped instance [TopologicalSpace S] : TopologicalSpace (R ⟶ S) :=
+  .induced (fun f ↦ f.hom : _ → R → S) inferInstance
+
+@[fun_prop]
+nonrec lemma continuous_apply [TopologicalSpace S] (x : R) :
+    Continuous (fun f : R ⟶ S ↦ f.hom x) :=
+  (continuous_apply x).comp continuous_induced_dom
+
+@[fun_prop]
+lemma continuous_precomp [TopologicalSpace T] (f : R ⟶ S) :
+    Continuous ((f ≫ ·) : (S ⟶ T) → (R ⟶ T)) :=
+  continuous_induced_rng.mpr ((Pi.continuous_precomp f.hom).comp continuous_induced_dom)
+
+/-- If `R ≅ S`, then `Hom(R, T)` is homeomorphc to `Hom(S, T)`. -/
+@[simps]
```

Comment:
Because you did not supply the `toFun` field explicitly, the apply lemma generated by `@[simps]` has an inaccessible variable name for the input. My instinct is that this should be avoided because you could imagine Lean auto-naming a variable with an inaccessible name when using the apply lemma. You can avoid this potential pitfall with `toFun x := _` I guess?